packages/core/src/RenderPipeline/MaskManager.ts

```diff
@@ -19,29 +20,57 @@ export class MaskManager {
     return (MaskManager._maskDecrementRenderQueue ||= new RenderQueue(RenderQueueType.Transparent));
   }
 
-  allSpriteMasks = new DisorderedArray<SpriteMask>();
-  preMaskLayer = 0;
+  private _allSpriteMasks = new DisorderedArray<SpriteMask>();
+  private _preMaskLayer = SpriteMaskLayer.Nothing;
 
   addSpriteMask(mask: SpriteMask): void {
-    mask._maskIndex = this.allSpriteMasks.length;
-    this.allSpriteMasks.add(mask);
+    mask._maskIndex = this._allSpriteMasks.length;
+    this._allSpriteMasks.add(mask);
   }
 
   removeSpriteMask(mask: SpriteMask): void {
-    const replaced = this.allSpriteMasks.deleteByIndex(mask._maskIndex);
+    const replaced = this._allSpriteMasks.deleteByIndex(mask._maskIndex);
     replaced && (replaced._maskIndex = mask._maskIndex);
     mask._maskIndex = -1;
   }
 
-  buildMaskRenderElement(
-    element: SubRenderElement,
+  drawMask(
+    context: RenderContext,
+    renderQueueType: RenderQueueType,
+    pipelineStageTagValue: string,
+    maskLayer: SpriteMaskLayer = SpriteMaskLayer.Nothing
+  ): void {
+    const incrementMaskQueue = MaskManager.getMaskIncrementRenderQueue();
+    incrementMaskQueue.renderQueueType = renderQueueType;
+    incrementMaskQueue.clear();
+    const decrementMaskQueue = MaskManager.getMaskDecrementRenderQueue();
+    decrementMaskQueue.renderQueueType = renderQueueType;
+    decrementMaskQueue.clear();
+
+    this._buildMaskRenderElement(maskLayer, incrementMaskQueue, decrementMaskQueue);
+
+    const engine = context.camera.engine;
+    const batcherManager = engine._batcherManager;
+    incrementMaskQueue.batch(engine._batcherManager);
+    batcherManager.uploadBuffer();
+    incrementMaskQueue.render(context, pipelineStageTagValue, StencilOperation.IncrementSaturate);
+    decrementMaskQueue.batch(engine._batcherManager);
+    batcherManager.uploadBuffer();
+    decrementMaskQueue.render(context, pipelineStageTagValue, StencilOperation.DecrementSaturate);
+  }
+
+  destroy(): void {
+    this._allSpriteMasks.length = 0;
```

Comment:
this._allSpriteMasks.garbageCollection();

Reply:
done